.github/workflows/build_and_test.yml

```diff
@@ -24,41 +25,59 @@ jobs:
       - uses: styfle/cancel-workflow-action@0.4.1
         with:
           access_token: ${{ secrets.GITHUB_TOKEN }}
-
+      
   build-test:
     runs-on: ubuntu-latest
-    container: "integritee/integritee-dev:0.1.9"
     steps:
       - uses: actions/checkout@v2
-      - name: init rust
-        run: rustup show
 
-      - name: Build
-        run:  env "MITIGATION-CVE-2020-0551=LOAD" "SGX_MODE=SW" make
-      - name: Test Service & Client
-        run: cargo test
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build Enclave Test Image
+        run:  docker build -t integritee-worker-enclave-test-${{ github.sha }} --target enclave-test -f build.Dockerfile .
+
       - name: Test Enclave # cargo test is not supported, see: https://github.com/apache/incubator-teaclave-sgx-sdk/issues/232
+        run: docker run --name ${{ env.BUILD_CONTAINER_NAME }} integritee-worker-enclave-test-${{ github.sha }}
+
+      - name: Build Cargo Test Image
+        run:  docker build -t integritee-worker-ctest-${{ github.sha }} --target cargo-test -f build.Dockerfile .
+
+      - name: Run Cargo Test
+        run: docker run --rm integritee-worker-ctest-${{ github.sha }}
+
+      - name: Build Deployable Image
+        run: docker buildx build -t integritee-worker-${{ github.sha }} --target deployed --output=type=docker,dest=/tmp/integritee-worker.tar -f build.Dockerfile .
```

Comment:
This image is a complete docker image in which a worker runs. In the future (with issue #735), we'll use this image for our integration test setup with docker-compose. Replaces the python script. For now, we just build the image and export it as artifact.